src/snowflake/snowpark/_internal/data_source/jdbc.py

```diff
@@ -0,0 +1,582 @@
+            as
+            $$
+
+            import java.sql.*;
+            import java.util.stream.Stream;
+            import java.util.ArrayList;
+            import java.util.List;
+            import com.snowflake.snowpark_java.types.*;
+            import com.snowflake.snowpark_java.types.SnowflakeSecrets;
+            import com.snowflake.snowpark_java.types.UsernamePassword;
+
+
+            class OutputRow {{
+
+                public String field_name;
+                public String jdbc_type;
+                public String java_type;
+                public int precision;
+                public int scale;
+                public boolean nullable;
+
+            }}
+
+            public class DataLoader{{
+                private final Connection conn;
+
+                private static Connection createConnection() {{
+                    try {{
+                        {self.generate_create_connection()}
+                    }} catch (Exception e) {{
+                        throw new RuntimeException("Failed to create JDBC connection: " + e.getMessage());
+                    }}
+                }}
+                public static Class<?> getOutputClass() {{
+                    return OutputRow.class;
+                }}
+
+                public DataLoader() {{
+                    this.conn = createConnection();
+                }}
+
+                public Stream<OutputRow> process(String query) {{
+                    List<OutputRow> list = new ArrayList<>();
+                     try {{
+                        PreparedStatement stmt = this.conn.prepareStatement(query);
+                        try (ResultSet rs = stmt.executeQuery()) {{
+                            ResultSetMetaData meta = rs.getMetaData();
+                            int columnCount = meta.getColumnCount();
+
+                            for (int i = 1; i <= columnCount; i++) {{
+                                OutputRow row = new OutputRow();
+
+                                row.field_name = meta.getColumnName(i);
+                                row.java_type = meta.getColumnClassName(i);
+                                try{{
+                                    row.jdbc_type = JDBCType.valueOf(meta.getColumnType(i)).toString();
+                                }} catch(Exception e){{
+                                    row.jdbc_type = null;
+                                }}
+                                row.precision = meta.getPrecision(i);
+                                row.scale = meta.getScale(i);
+                                row.nullable = meta.isNullable(i) == ResultSetMetaData.columnNullable;
+                                list.add(row);
+                            }}
+                        }}
+                    }} catch (SQLException e) {{
+                        throw new RuntimeException("SQL error: " + e.getMessage(), e);
+                    }}
+
+                    return list.stream();
```

Comment:
IIUC, this loads all data into local first, which is likely to cause OOM if the data is large.

I think we should use Stream.generate like: https://medium.com/@orellabac/ingest-external-data-into-snowflake-with-snowpark-and-jdbc-eb487b61078c